jupyterhub/values.yaml

```diff
@@ -350,6 +350,13 @@ scheduling:
 
 prePuller:
   annotations: {}
+  resources:
+    requests:
+      cpu: 0
+      memory: 0
+    limits:
+      cpu: 10m
+      memory: 10Mi
```

Comment:
```suggestion
      memory: 1Mi
```

Reply:
## About 0 requests

These were the resources reported back from `kubectl get pod -o yaml` after trying to set the resources like described. I suspect that KubeSpawner may have a logic checking for a truthy value of the cpu request and leaving it out if not set or similar could be an issue though.

```yaml
    resources:
      limits:
        cpu: 10m
        memory: 10Mi
      requests:
        cpu: "0"
        memory: "0"
```

Relevant to this is that I've now learned that `LimitRange` resources can specify a `maxLimitRequestRatio` which would enforce the resource type influenced to not have any field set to zero. Do we want to avoid this by having a request of 1?

## Reducing limits further

We can do this, but are you sure that the containers will stay below 1Mi of memory usage? My main concern is that I'll have concern about this when debugging some other issue and spend time thinking about if this narrow limit could have caused problems.

I decided to try using a extremely narrow limit of 1m / 10Mi and found that in my k8s cluster with Calico etc the pod with 1m / 1Mi limits got stuck in `ContainerCreating` while the 10m / 10Mi limited pod got started successfully.

![image](https://user-images.githubusercontent.com/3837114/92211171-4e06ef00-ee90-11ea-8bb8-610324c7472d.png)

The issues were the following on the 1m / 10 Mi pod.

```
Events:
  Type     Reason                  Age                   From                                                Message
  ----     ------                  ----                  ----                                                -------
  Normal   Scheduled               3m18s                 default-scheduler                                   Successfully assigned default/tmp-6f699cc94f-62sm6 to gke-nh-2020-core-standard-2-b80a0890-kfvt
  Warning  FailedCreatePodSandBox  104s (x3 over 2m47s)  kubelet, gke-nh-2020-core-standard-2-b80a0890-kfvt  Failed create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "tmp-6f699cc94f-62sm6": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:319: getting the final child's pid from pipe caused \"read init-p: connection reset by peer\"": unknown
  Warning  FailedCreatePodSandBox  33s (x2 over 68s)     kubelet, gke-nh-2020-core-standard-2-b80a0890-kfvt  Failed create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "tmp-6f699cc94f-62sm6": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"read init-p: connection reset by peer\"": unknown
  Normal   SandboxChanged          32s (x5 over 2m46s)   kubelet, gke-nh-2020-core-standard-2-b80a0890-kfvt  Pod sandbox changed, it will be killed and re-created.
```

I think the newly introduced [Pod Overhead](https://kubernetes.io/docs/concepts/configuration/pod-overhead/) could mitigate this and allow us to have extremely low limits, but since that is a lot of extra infrastructure, I suggest we stick with the current 10 m / 10 Mi to be safe.

Reply:
@yuvipanda what do you think, merge as it is?

Reply:
I think we should actually remove the limits. On my cluster (AKS), setting this to 10Mi causes the pods to constantly crash as well. Why don't we just remove the limits?

Reply:
@yuvipanda ah yes deal. Are we unlimited by leaving them out?